src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -222,6 +222,17 @@ private ScriptEngineService getScriptEngineService(String lang) {
         return scriptEngineService;
     }
 
+    private ScriptEngineService getScriptEngineServiceForFileExt(String fileExtension) {
+        for (ScriptEngineService engineService : scriptEngines.values()) {
+            for (String s : engineService.extensions()) {
+                if (s.equals(fileExtension)) {
+                    return engineService;
+                }
+            }
+        }
+        return null;
```

Comment:
done, also noticed that we were closing multiple times the same engine pontentially, fixed that too.